lib/modules/hosts/giphy.js

```diff
@@ -1,27 +1,35 @@
+import { ajax } from '../../environment';
+import logo from '../../images/hosts/giphy-logo.png';
+
 export default {
 	moduleID: 'giphy',
 	name: 'giphy',
 	domains: ['giphy.com'],
-	logo: '//giphy.com/static/img/favicon.png',
-	detect: ({ pathname }) => (/^\/gifs\/(?:\w+-)*(.*?)(\/html5)?$/i).exec(pathname),
-	handleLink(href, [, id, isHtml5]) {
+	logo,
+	detect: ({ pathname }) => (/^(?:\/gifs|\/media|)\/(?:\w+-)*([^/.]*)(\/html5)?.*$/i).exec(pathname),
```

Comment:
Nit: `.*$` at the end of the regex is redundant; it can be removed. But to make the regex a bit more precise, I think we should replace it with `(?:\/|$)`

Is the trailing `|` in `(?:\/gifs|\/media|)` intended? If so, do you have an example link? If not, we can make that a bit cleaner: `\/(?:gifs|media)`

And while we're at it might as well replace `([^/.]*)` with `([^/.]+)` since an empty id isn't very useful (I'm not sure why it was this way originally).